pkg/auth/auth.go

```diff
@@ -79,7 +92,15 @@ func resolve(discoveryCli discovery.DiscoveryInterface, groupVersion, kind strin
 }
 
 func (u *UserAuth) canPerform(verb, group, resource, namespace string) (bool, error) {
+	meta := metav1.ObjectMeta{}
+	// In the test environment, fake API behaviour adding a name to the resource
+	if testEnv {
```

Comment:
Wouldn't it be better to create an interface for actions on UserAuth and have a fake implementation of it. I'm not a fan of mixing in testing logic here.

You can take a look at the interface for Datastore: https://github.com/kubeapps/common/blob/master/datastore/datastore.go and the Mock implementation: https://github.com/kubeapps/common/blob/master/datastore/mockstore/mockstore.go.

Reply:
yes, well, I can wrap the methods that calls to the K8s API and make an interface of it.